design/observerability-2.md

```diff
@@ -0,0 +1,196 @@
+# Observability #2
+
+This document deprecates [the previous version of Observability](observability.md)
+
+* **Status**: Draft #1
+* **JIRA**: [KEYCLOAK-8288](https://issues.jboss.org/browse/KEYCLOAK-8288)
+* **Healthcheck API JIRA**: [KEYCLOAK-12398](https://issues.redhat.com/browse/KEYCLOAK-12398)
+
+## Glossary
+
+- SRE - Site Reliability Engineering
+- SLI - Service Layer Indicator - basically a metric
+- SLO - An assumption about SLI. The SLI should be between the lower bound and the upper bound. [Here’s a good example](https://landing.google.com/sre/workbook/chapters/slo-document/).
+- SLA - Service Level Agreement - what are the SLOs written into an agreement between services (or Red Hat and end users for that matter).
+- Alert - A warning triggered when metrics went beyond a level defined in SLO.
+[Here’s the OpenShift Dedicated SLA (go to page 9)](https://www.redhat.com/licenses/Appendix_4_Online_Global_20200406.pdf).
+- Prometheus - A time-series database. We store metrics there.
+- Grafana - A service that presents diagrams based on metrics from Prometheus.
+- EFK Stack - Elsticsearch, Fluentd and Kibana. A typical log-collecting stack.
+- CLO - Cluster Logging Operator
+- CR - Custom Resource. An API provided by Keycloak Operator for the context of this document.
+
+## Motivation
+
+Managing Keycloak cluster requires involving a dedicated SRE (Site Reliability Engineering) Team. Such a team needs to
+be able to:
+
+* Diagnose if Keycloak cluster is up and running
+* Have sufficient information to diagnose issues
+* Have enough data and metrics to foresee problems that will happen in the near future
+
+Typically, an SRE Team uses dashboards and alerts. Both of them are created based on metrics exported by a Service
+(in this case, the Service is both Keycloak and Keycloak Operator).
+
+Based on metrics, dashboards and alerts, it is possible to define SLA - an agreement between a Service Provider and a
+Service Consumer.
+
+## Scope
+
+What's in the scope of this design:
+- Designing a new Healcheck API. This has been covered in a separate design document. See [KEYCLOAK-12398](https://issues.redhat.com/browse/KEYCLOAK-12398)
+- Designing and implementing a dashboard for the SRE Team.
+- Defining SLIs and SLOs as well as proposing template SLAs for a managed Keycloak cluster.
+- Defining metrics required by the proposed dashboard as well as designing a mechanism for exporting them to prometheus.
+- Design logging improvements for better error tracking.
+
+What's out of the scope:
+- Traceability - tracking a user across the whole authorization flow. At the moment we don't have any requirements
+in this direction. If we get such a requirement, we will track this progress in a separate epic and design.
```

Comment:
We do have "requirements in this direction", so just write that it is out of scope at this stage, and will be covered in a separate design/epic in the future

Reply:
Ok, will do that. I'll also find a Tractability epic JIRA and link it here.

Reply:
Updated.